clang/lib/Sema/SemaInit.cpp

```diff
@@ -6457,6 +6472,18 @@ void InitializationSequence::InitializeFrom(Sema &S,
     }
   }
 
+  if (!S.getLangOpts().CPlusPlus &&
+      Kind.getKind() == InitializationKind::IK_Default) {
+    RecordDecl *Rec = DestType->getAsRecordDecl();
+    if (Rec && Rec->hasUninitializedExplicitInitFields()) {
+      VarDecl *Var = dyn_cast_or_null<VarDecl>(Entity.getDecl());
+      if (Var && !Initializer) {
+        S.Diag(Var->getLocation(), diag::warn_field_requires_explicit_init)
+            << true << Rec;
```

Comment:
```suggestion
            << /* Var-in-Record */1 << Rec;
```

Select takes an integral result, so we should pass it one, rather than count on bool->int conversions

Reply:
I don't doubt it :/  We've got quite a bit of 'uncleaned up' code around as we've learned the downsides of allowing things like this that have never caused a problem, so have never been cleaned up.